ts/sagas/startup/watchApplicationActivitySaga.ts

```diff
@@ -1,106 +1,57 @@
-    const nowMillis = new Date().getTime();
-    const timeElapsedMillis = lastUpdateAtMillis
-      ? nowMillis - lastUpdateAtMillis
-      : nowMillis;
-
     if (lastState !== "background" && newApplicationState === "background") {
-      // The app is going into background
-      // Save the navigation state so we can restore it later when the app come
-      // back to the active state
-      yield call(saveNavigationStateSaga);
-
-      // Make sure that when the app come back active, the BackgrounScreen
-      // gets loaded first
-      // FIXME: not that this creates a quick blue flash in case after restoring
-      //        the app we don't ask a PIN
-      yield put(navigateToBackgroundScreen);
+      // Start the background timer
+      identificationBackgroundTimer = yield fork(
+        startIdentificationBackgroundTimer,
+        backgroundActivityTimeoutMillis
+      );
     } else if (lastState === "background" && newApplicationState === "active") {
-      // The app is coming back active after being in background
-      if (timeElapsedMillis > backgroundActivityTimeoutMillis) {
-        // If the app has been in background state for more than the timeout,
-        // re-initialize the app from scratch
-        yield put(startApplicationInitialization());
-      } else {
-        // Check if we have a pending notification message
-        const pendingMessageState: PendingMessageState = yield select<
-          GlobalState
-        >(pendingMessageStateSelector);
-        if (pendingMessageState) {
-          // We have a pending notification message to handle
-          const messageId = pendingMessageState.id;
-          // Remove the pending message from the notification state
-          yield put(clearNotificationPendingMessage());
-          // Navigate to message details screen
-          yield put(navigateToMessageDetailScreenAction({ messageId }));
-          // Push the MAIN navigator in the history to handle the back button
-          const navigationState: NavigationState = yield select<GlobalState>(
-            navigationStateSelector
-          );
-          yield put(
-            navigationHistoryPush(
-              AppNavigator.router.getStateForAction(
-                NavigationActions.back(),
-                navigationState
-              )
-            )
-          );
-        } else {
-          // Or else, just navigate back to the screen we were at before
-          // going into background
-          yield put(NavigationActions.back());
-        }
+      // Cancel the background timer if running
+      if (identificationBackgroundTimer) {
+        yield cancel(identificationBackgroundTimer);
+        identificationBackgroundTimer = undefined;
       }
     }
 
-    // Update the last state and update time
+    // Update the last state
     lastState = newApplicationState;
-    lastUpdateAtMillis = nowMillis;
   });
 }
+
+function* startIdentificationBackgroundTimer(
```

Comment:
this function is only two instructions and it's only called once, why not inline it?